tools/mason/MasonInit.chpl

```diff
@@ -37,43 +37,90 @@ Initialises a library project in a project directory
 */
 proc masonInit(args) throws {
   try! {
-    var name = '';
+    var dirName = '';
     var show = false;
+    var packageName = '';
+    var countArgs = 2;
     for arg in args[2..] {
+      countArgs += 1;
       if arg == '-h' || arg == '--help' {
         masonInitHelp();
         exit();
       }
       else if arg == '--show' {
         show = true;
       }
+      else if arg.startsWith('--name') {
+        if arg.startsWith('--name=') {
+          var res = arg.split("=");
+          packageName = res[2];
+          dirName = args[2];
+        }
+        else {
+          packageName = args[countArgs];
+          dirName = args[2];
+        }
+      }
       else {
-        name = arg;
+        if packageName.length == 0 then {
+          dirName = arg;
+          packageName = dirName;
+        }
       }
     }
-    if name == '' {
+    if dirName.startsWith('--name') then dirName = '';
+    if dirName == '' {
       const cwd = getEnv("PWD");
-      const name = basename(cwd);
+      var name = basename(cwd); 
       const path = '.';
-      if validatePackageName(name) {
-        validateMasonFile(path, name, show);
-        var isInitialized = validateInit(path, show);
-        if isInitialized > 0 then
-          writeln("Initialized new library project: " + name);
+      if dirName != packageName then {
+        createModule(path, packageName, show);
+        if validatePackageName(packageName) {
+          validateMasonFile(path, packageName, show);
+          var isInitialized = validateInit(path, name=packageName, true, show);
+          if isInitialized > 0 then
+          writeln("Initialized new library project: " + basename(cwd));
+        }
+      }
+      else {
+        var resName = validatePackageNameChecks(path, name); 
+        var isNamesDiff = false;
+        if resName != name then {name = resName; isNamesDiff = true;} 
+        if resName == name {
+          validateMasonFile(path, name, show);
+          var isInitialized = validateInit(path, name, isNamesDiff, show);
+          if isInitialized > 0 then
+          writeln("Initialized new library project: " + basename(cwd));
+        }
       }
     }
     else {
       // if the target directory in path doesnt exist, throw error
       // if target directory exists, check for files && validate
       // create folders and toml file without overwriting anything
       // if TOML file exists, check for values in it and validate
-      const path = name;
+      const path = dirName;
       if isDir(path) {
-        if validatePackageName(name) {
-          validateMasonFile(path, basename(path), show);
-          var isInitialized = validateInit(path, show);
-          if isInitialized > 0 then
+        if dirName != packageName then {
```

Comment:
We shouldn't need to branch here. The calls should be the same if regardless of whether `dirName == packageName` or not.